samples/tutorial/tutorial-2-complete/src/main/java/workflow/tutorial/RootWorkflow.kt

```diff
@@ -0,0 +1,73 @@
+package workflow.tutorial
+
+import com.squareup.workflow1.Snapshot
+import com.squareup.workflow1.StatefulWorkflow
+import com.squareup.workflow1.action
+import com.squareup.workflow1.renderChild
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.backstack.BackStackScreen
+import com.squareup.workflow1.ui.backstack.toBackStackScreen
+import workflow.tutorial.RootWorkflow.State
+import workflow.tutorial.RootWorkflow.State.Todo
+import workflow.tutorial.RootWorkflow.State.Welcome
+import workflow.tutorial.TodoListWorkflow.ListProps
+
+@OptIn(WorkflowUiExperimentalApi::class)
+object RootWorkflow : StatefulWorkflow<Unit, State, Nothing, BackStackScreen<*>>() {
+
+  sealed class State {
+    object Welcome : State()
+    data class Todo(val name: String) : State()
+  }
+
+  override fun initialState(
+    props: Unit,
+    snapshot: Snapshot?
+  ): State = Welcome
+
+  @OptIn(WorkflowUiExperimentalApi::class)
+  override fun render(
+    props: Unit,
+    state: State,
+    context: RenderContext
+  ): BackStackScreen<*> {
+
+    // Our list of back stack items. Will always include the "WelcomeScreen".
+    val backstackScreens = mutableListOf<Any>()
+
+    // Render a child workflow of type WelcomeWorkflow. When renderChild is called, the
+    // infrastructure will create a child workflow with state if one is not already running.
+    val welcomeScreen = context.renderChild(WelcomeWorkflow) { output ->
+      // When WelcomeWorkflow emits LoggedIn, turn it into our login action.
+      login(output.name)
+    }
+    backstackScreens += welcomeScreen
+
+    when (state) {
+      // When the state is Welcome, defer to the WelcomeWorkflow.
+      is Welcome -> {
+        // We always add the welcome screen to the backstack, so this is a no op.
+      }
+
+      // When the state is Todo, defer to the TodoListWorkflow.
+      is Todo -> {
+        backstackScreens += context.renderChild(TodoListWorkflow, ListProps(name = state.name)) {
+          logout()
+        }
+      }
+    }
+
+    // Finally, return the BackStackScreen with a list of BackStackScreen.Items
+    return backstackScreens.toBackStackScreen()
+  }
+
+  override fun snapshotState(state: State): Snapshot? = null
+
+  private fun login(name: String) = action {
+    state = Todo(name)
+  }
+
+  private fun logout() = action {
```

Comment:
Nit: I'd normally make a no-args action a val.

Reply:
I did everything as functions because it looks more consistent. I don't really care though, I can do vals but i'll do it in a follow up so I can change everything at once.

Reply:
Done.